subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/modulecache/ModuleMetadataCache.java

```diff
@@ -24,11 +24,16 @@
 import javax.annotation.Nullable;
 
 public interface ModuleMetadataCache {
-    CachedMetadata cacheMissing(ModuleComponentRepository repository, ModuleComponentIdentifier id);
+    CachedMetadata cacheMissing(ModuleMetadataDetails details);
```

Comment:
I'm not convinced that the introduction of this `details` object is an improvement. I'm not sure what it represents, except for a parameter object for this particular method. This seems overkill for 2 parameters, and the cost is that it makes the API for `ModuleMetadataCache.getCachedModuleDescriptor()` less obvious.

If you're convinced this abstraction is worth it, then we should do the same for the other cache interfaces, to maintain the same pattern.

Reply:
I introduced it when I had to add a 3rd parameter (which has been removed since then), so I can revert the change.